api/views.py

```diff
@@ -95,7 +95,18 @@ class AssetViewSet(ModelViewSet):
     def get_queryset(self):
         user = self.request.user
         asset_assignee = AssetAssignee.objects.filter(user=user).first()
-        queryset = Asset.objects.filter(assigned_to=asset_assignee)
+        query_filter = {}
+        has_filter = False
+        for field in self.request.query_params:
+            if field == 'serial_number' or field == 'asset_code':
+                query_filter[field] = self.request.query_params.get(field)
+                has_filter = True
+
+        queryset = Asset.objects.filter(assigned_to=asset_assignee, **query_filter)
```

Comment:
i believe we should include ` **query_filter` only if `has_filter ` is `True`

and 1 last change.. we should allow security users to query this endpoint for all assets. ie if it's a security user making the request they can see assets belonging to other users (we shouldn't check `assigned_to ` for requests from security users)

Reply:
alright